### PR TITLE
avm2: Don't overwrite existing entry in `export_class`

### PR DESCRIPTION
**Pull request by Aaron1011**
This makes the behavior consistent with `export_definition`

### CODE REVIEW

**Review by relrelb — APPROVED** [review]

**Comment by Lord-McSweeney** [comment]
Can you add a test for this?

File: core/src/avm2/domain.rs [hunk 1]
```diff
@@ -245,6 +245,9 @@ impl<'gc> Domain<'gc> {
     }
 
     pub fn export_class(&self, class: GcCell<'gc, Class<'gc>>, mc: MutationContext<'gc, '_>) {
```
Comment (iwannabethedev) [inline core/src/avm2/domain.rs:247]: (Nit: Maybe refactor the key value into its own variable, like:

```rust
        let class_name = class.read().name();
        if self.0.read().classes.contains_key(class_name) {
            return;
        }
        self.0.write(mc).classes.insert(class_name, class);
```

?)

File: core/src/avm2/domain.rs [hunk 2]
```diff
@@ -245,6 +245,9 @@ impl<'gc> Domain<'gc> {
...
+        if self.0.read().classes.contains_key(class.read().name()) {
+            return;
```
Comment (iwannabethedev) [inline core/src/avm2/domain.rs:249]: I do not know ActionScript, and I do not quite understand this code either (not because it is unclear, the code seems clear, but because I do not know much about this part of the codebase), but, if I had to guess, if we "include" a class/type (like `MyDuplicate`), and we then "include" it some other way (using for instance `Loader.load()`), we do not want to "include" what is the same class again, since it has already been "included", correct? If so, the change seems good. Though, is `name()` qualified with a package path (if that makes sense)? Like, if I had two different types `package1.MyDuplicate` and `package2.MyDuplicate`, would it handle that correctly? But I do not know ActionScript and I do not quite understand this part of the codebase.

**Comment by iwannabethedev** [comment]
I do not understand this part of the codebase, and I know very little about ActionScript, so someone else should review it as well, I think.

File: core/src/avm2/domain.rs (hunk 1 above)
Comment (relrelb) [reply core/src/avm2/domain.rs:247]: I think it's readable enough as-is.

File: core/src/avm2/domain.rs [hunk 3]
```diff
@@ -245,6 +245,9 @@ impl<'gc> Domain<'gc> {
     }
```
[the remark this answers is not in the document]
Comment (relrelb) [reply core/src/avm2/domain.rs:246]: Actually that would be nice. Since I want to merge this PR already, would you like you add such a comment in a separate PR?

File: core/src/avm2/domain.rs (hunk 2 above)
Comment (relrelb) [reply core/src/avm2/domain.rs:249]: You can see how the fix is relevant in the recently added test. Nothing actionable in this thread, so resolving.

**Review by relrelb — APPROVED** [review]

File: core/src/avm2/domain.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (iwannabethedev) [reply core/src/avm2/domain.rs:246]: That sounds alright, though I am not exactly certain what it should say :smile: .

Maybe:

```rust
/// Export a class from a script into the current application domain.
```
? But I would feel more confident if someone who actually knows this part of the codebase made (or approved of) the comment.

File: core/src/avm2/domain.rs (hunk 2 above)
Comment (iwannabethedev) [reply core/src/avm2/domain.rs:249]: That is not actually what I am asking about, sorry if I am being unclear. I did see the test, and I am referring to it in the comment. What I first asked about was if my understanding of the changes (and the test) is correct, and then I asked if a certain corner case is handled correctly. Which I am still in doubt about - do you know if `name()`  is qualified with a package path (if that makes sense)?
